Plugins/Public/munitioncntl/Main.cpp

```diff
@@ -0,0 +1,205 @@
+			if (ini.is_header("NoTrackingAlert"))
+			{
+				while (ini.read_value())
+				{
+					if (ini.is_value("MissileArch"))
+					{
+						setNoTrackingAlertProjectiles.insert(CreateID(ini.get_value_string(0)));
+					}
+				}
+			}
+			else if (ini.is_header("TrackingBlacklist"))
+			{
+				uint missileArch = 0;
+				uint blacklistedTrackingTypesBitmap = 0;
+				while (ini.read_value())
+				{
+					if (ini.is_value("MissileArch"))
+					{
+						missileArch = CreateID(ini.get_value_string(0));
+					}
+					else if (ini.is_value("Type"))
+					{
+						string typeStr = ToLower(ini.get_value_string(0));
+						if (typeStr.find("fighter") != string::npos)
+							blacklistedTrackingTypesBitmap |= OBJ_FIGHTER;
+						if (typeStr.find("freighter") != string::npos)
+							blacklistedTrackingTypesBitmap |= OBJ_FREIGHTER;
+						if (typeStr.find("transport") != string::npos)
+							blacklistedTrackingTypesBitmap |= OBJ_TRANSPORT;
+						if (typeStr.find("gunboat") != string::npos)
+							blacklistedTrackingTypesBitmap |= OBJ_GUNBOAT;
+						if (typeStr.find("cruiser") != string::npos)
+							blacklistedTrackingTypesBitmap |= OBJ_CRUISER;
+						if (typeStr.find("capital") != string::npos)
+							blacklistedTrackingTypesBitmap |= OBJ_CAPITAL;
+						if (typeStr.find("guided") != string::npos)
+							blacklistedTrackingTypesBitmap |= OBJ_GUIDED;
+						if (typeStr.find("mine") != string::npos)
+							blacklistedTrackingTypesBitmap |= OBJ_MINE;
+					}
+				}
+				mapTrackingByObjTypeBlacklistBitmap[missileArch] = blacklistedTrackingTypesBitmap;
+			}
+		}
+		ini.close();
+	}
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+//Functions
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+void ProcessGuided(FLPACKET_CREATEGUIDED& createGuidedPacket)
+{
+	uint ownerType;
+	pub::SpaceObj::GetType(createGuidedPacket.iOwner, ownerType);
+	if (!(ownerType & (OBJ_FIGHTER | OBJ_FREIGHTER | OBJ_TRANSPORT | OBJ_GUNBOAT | OBJ_CRUISER | OBJ_CAPITAL))) //GetTarget throws an exception for non-ship entities.
+		return;
+	uint targetId;
+	pub::SpaceObj::GetTarget(createGuidedPacket.iOwner, targetId);
+
+	TRACKING_STATE tracking = TRACK_ALERT;
+
+	if (!targetId) // prevent missiles from tracking cloaked ships, and missiles sticking targeting to last selected target
+	{
+		tracking = NOTRACK_NOALERT;
+	}
+	else if (setNoTrackingAlertProjectiles.count(createGuidedPacket.iMunitionId)) // for 'dumbified' seeker missiles, disable alert, used for flaks and snub dumbfires
+	{
+		tracking = TRACK_NOALERT;
+	}
+	else if (mapTrackingByObjTypeBlacklistBitmap.count(createGuidedPacket.iMunitionId)) // disable tracking for selected ship types
+	{
+		uint targetType;
+		pub::SpaceObj::GetType(createGuidedPacket.iTargetId, targetType);
+		const auto& blacklistedShipTypeTargets = mapTrackingByObjTypeBlacklistBitmap.at(createGuidedPacket.iMunitionId);
```

Comment:
Do we validate that this is in the list?

Reply:
Within the if above it, using the .count() method.